app/src/test/java/com/hover/stax/data/remote/AuthApiTest.kt

```diff
@@ -0,0 +1,319 @@
+            )
+        }
+
+        val api = StaxApi(
+            KtorClientFactory(
+                DefaultTokenProvider(testDataStore),
+                environmentProvider
+            ).create(mockEngine), environmentProvider
+        )
+
+        val actual = runBlocking { api.authorize(authRequest) }
+
+        assertThat(actual).isEqualTo(authResponse)
+    }
+
+    @Test
+    fun `test token request is successful when authorization code is correct`() {
+        val tokenRequest = fixture<TokenRequest>()
+        val tokenResponse = TokenResponse(
+            accessToken = "eyJpc3MiOiJTdGF4IE1vYmlsZSIsImlhdCI6MTY2MTc4NTkxMSwianRpIjoiZDFiYmEwZGItOWNiOC00OWUxLWEzNTItODk3NzYxMDhjYWZkIiwidXNlciI6IntcImRhdGFcIjp7X",
+            refreshToken = "A3_rBRCZUtL3i0h-y2_HWtw1icW9ZUaeH9Fq7R42GXg",
+            scope = "write",
+            createdAt = 1661785911,
+            tokenType = "Bearer",
+            expiresIn = 7200
+        )
+        val response = """{
+                "access_token" : "eyJpc3MiOiJTdGF4IE1vYmlsZSIsImlhdCI6MTY2MTc4NTkxMSwianRpIjoiZDFiYmEwZGItOWNiOC00OWUxLWEzNTItODk3NzYxMDhjYWZkIiwidXNlciI6IntcImRhdGFcIjp7X",
+                "refresh_token" : "A3_rBRCZUtL3i0h-y2_HWtw1icW9ZUaeH9Fq7R42GXg",
+                "scope" : "write",
+                "created_at" : 1661785911,
+                "token_type" : "Bearer",
+                "expires_in" :  7200
+        }""".trimIndent()
+
+        val mockEngine = MockEngine {
+            respond(
+                content = response,
+                status = HttpStatusCode.OK,
+                headersOf(HttpHeaders.ContentType, "application/json")
+            )
+        }
+
+        val api = StaxApi(
+            KtorClientFactory(
+                DefaultTokenProvider(testDataStore),
+                environmentProvider
+            ).create(mockEngine), environmentProvider
+        )
+
+        val actual = runBlocking { api.fetchToken(tokenRequest) }
+
+        assertThat(actual).isEqualTo(tokenResponse)
+    }
+
+    @Test
+    fun `test token revoke is successful when revoke token is correct`() {
+        val tokenRevoke = fixture<RevokeTokenRequest>()
+
+        val response = """{}""".trimIndent()
+
+        val mockEngine = MockEngine {
+            respond(
+                content = response,
+                status = HttpStatusCode.OK,
+                headersOf(HttpHeaders.ContentType, "application/json")
+            )
+        }
+
+        val api = StaxApi(
+            KtorClientFactory(
+                DefaultTokenProvider(testDataStore),
+                environmentProvider
+            ).create(mockEngine), environmentProvider
+        )
+
+        val actual = runBlocking { api.revokeToken(tokenRevoke) }
+
+        assertThat(actual).isInstanceOf(HttpResponse::class.java)
+    }
+
+    @Test
+    fun `test upload user is successful when passing correct user details`() {
+        val userDTO = fixture<UserUploadDto>()
+        val staxUserDto = StaxUserDto(
```

Comment:
I observed the same userDTO is required twice. Can we have a private getMockUser()? function

Reply:
Yes, we can do that 

Fixed